File: app/src/main/java/com/intkhabahmed/smartnotes/viewmodels/NotesViewModel.java
package com.intkhabahmed.smartnotes.viewmodels;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.intkhabahmed.smartnotes.database.NoteRepository;
import com.intkhabahmed.smartnotes.models.Note;

import java.util.List;

public class NotesViewModel extends ViewModel {
    private LiveData<List<Note>> notes;

    NotesViewModel(String noteType, int trashed) {
        if (notes == null) {
            setNotes(noteType, trashed);
        }
    }

    private void setNotes(String noteType, int trashed) {
        notes = NoteRepository.getInstance().getNotesByTypeAndAvailability(noteType, trashed);
    }

    public LiveData<List<Note>> getNotes() {
        return notes;
    }
}

File: app/src/main/java/com/intkhabahmed/smartnotes/viewmodels/NotesViewModel.kt
package com.intkhabahmed.smartnotes.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.paging.LivePagedListBuilder
import androidx.paging.PagedList

import com.intkhabahmed.smartnotes.database.NoteRepository
import com.intkhabahmed.smartnotes.models.Note

class NotesViewModel internal constructor(noteType: String, trashed: Int) : ViewModel() {
    lateinit var notes: LiveData<PagedList<Note>>

    init {
        setNotes(noteType, trashed)
    }

    fun setNotes(noteType: String, trashed: Int) {
        notes = LivePagedListBuilder(
                NoteRepository.instance!!.getNotesByTypeAndAvailability(noteType, trashed),
                PagedList.Config.Builder().setPageSize(10).setInitialLoadSizeHint(20).setEnablePlaceholders(true).build()
        ).build()
    }
}
